PS C:\Users\Cristian\Downloads> cd ..
PS C:\Users\Cristian> cd djangogirls
PS C:\Users\Cristian\djangogirls> myvenv\Scripts\activate
(myvenv) PS C:\Users\Cristian\djangogirls> python manage.py runserver
Watching for file changes with StatReloader
Performing system checks...

Exception in thread django-main-thread:
Traceback (most recent call last):
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2288.0_x64__qbz5n2kfra8p0\Lib\threading.py", line 1045, in _bootstrap_inner  
    self.run()
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2288.0_x64__qbz5n2kfra8p0\Lib\threading.py", line 982, in run
    self._target(*self._args, **self._kwargs)
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\utils\autoreload.py", line 64, in wrapper
    fn(*args, **kwargs)
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\core\management\commands\runserver.py", line 118, in inner_run
    self.check(display_num_errors=True)
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\core\management\base.py", line 419, in check
    all_issues = checks.run_checks(
                 ^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\core\checks\registry.py", line 76, in run_checks
    new_errors = check(app_configs=app_configs, databases=databases)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\core\checks\urls.py", line 40, in check_url_namespaces_unique
    all_namespaces = _load_all_namespaces(resolver)
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\core\checks\urls.py", line 57, in _load_all_namespaces
    url_patterns = getattr(resolver, 'url_patterns', [])
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\utils\functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
                                         ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\urls\resolvers.py", line 602, in url_patterns
    patterns = getattr(self.urlconf_module, "urlpatterns", self.urlconf_module)
                       ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\utils\functional.py", line 48, in __get__
    res = instance.__dict__[self.name] = self.func(instance)
                                         ^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\urls\resolvers.py", line 595, in urlconf_module
    return import_module(self.urlconf_name)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Program Files\WindowsApps\PythonSoftwareFoundation.Python.3.11_3.11.2288.0_x64__qbz5n2kfra8p0\Lib\importlib\__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1147, in _find_and_load_unlocked 
  File "<frozen importlib._bootstrap>", line 690, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 936, in exec_module     
  File "<frozen importlib._bootstrap_external>", line 1074, in get_code       
  File "<frozen importlib._bootstrap_external>", line 1004, in source_to_code 
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed  File "C:\Users\Cristian\djangogirls\mysite\urls.py", line 21  File "C:\Users\Cristian\djangogirls\mysite\urls.py", line 21
    path(''include('blog.urls')),
         ^^^^^^^^^^^^^^^^^^^^^^
SyntaxError: invalid syntax. Perhaps you forgot a comma?
C:\Users\Cristian\djangogirls\mysite\urls.py changed, reloading.
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
February 25, 2024 - 21:58:54
Django version 3.2.24, using settings 'mysite.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CTRL-BREAK.
(myvenv) PS C:\Users\Cristian\djangogirls> python manage.py runserver
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
February 25, 2024 - 22:00:08
Django version 3.2.24, using settings 'mysite.settings'
Starting development server at http://127.0.0.1:8000/  
Quit the server with CTRL-BREAK.
Internal Server Error: /
Traceback (most recent call last):
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\core\handlers\exception.py", line 47, in inner    
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\core\handlers\base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\blog\views.py", line 5, in post_list
    return render(request, 'blog/post_list.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\shortcuts.py", line 19, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string 
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\template\loader.py", line 19, in get_template     
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: blog/post_list.html
[25/Feb/2024 22:00:16] "GET / HTTP/1.1" 500 77995
Internal Server Error: /
Traceback (most recent call last):
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\core\handlers\exception.py", line 47, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\core\handlers\base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\blog\views.py", line 5, in post_list
    return render(request, 'blog/post_list.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\shortcuts.py", line 19, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\template\loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: blog/post_list.html
[25/Feb/2024 22:03:23] "GET / HTTP/1.1" 500 77995
Internal Server Error: /
Traceback (most recent call last):
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\core\handlers\exception.py", line 47, in inner
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\core\handlers\base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\blog\views.py", line 5, in post_list
    return render(request, 'blog/post_list.html', {})
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\shortcuts.py", line 19, in render
    content = loader.render_to_string(template_name, context, request, using=using)
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\template\loader.py", line 61, in render_to_string
    template = get_template(template_name, using=using)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\template\loader.py", line 19, in get_template
    raise TemplateDoesNotExist(template_name, chain=chain)
django.template.exceptions.TemplateDoesNotExist: blog/post_list.html
[25/Feb/2024 22:03:47] "GET / HTTP/1.1" 500 78001
(myvenv) PS C:\Users\Cristian\djangogirls> python manage.py runserver
Watching for file changes with StatReloader

System check identified no issues (0 silenced).
February 25, 2024 - 22:04:08
Django version 3.2.24, using settings 'mysite.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CTRL-BREAK.
[25/Feb/2024 22:04:11] "GET / HTTP/1.1" 200 0
(myvenv) PS C:\Users\Cristian\djangogirls> python manage.py runserver
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
February 25, 2024 - 22:04:48
Django version 3.2.24, using settings 'mysite.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CTRL-BREAK.
[25/Feb/2024 22:04:52] "GET / HTTP/1.1" 200 87
(myvenv) PS C:\Users\Cristian\djangogirls> python manage.py runserver
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
February 25, 2024 - 22:05:30
Django version 3.2.24, using settings 'mysite.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CTRL-BREAK.
(myvenv) PS C:\Users\Cristian\djangogirls> python manage.py runserver
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
February 25, 2024 - 22:06:34
Django version 3.2.24, using settings 'mysite.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CTRL-BREAK.
[25/Feb/2024 22:06:41] "GET / HTTP/1.1" 200 981
(myvenv) PS C:\Users\Cristian\djangogirls> git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
        modified:   blog/views.py
        modified:   mysite/urls.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        blog/templates/
        blog/urls.py

no changes added to commit (use "git add" and/or "git commit -a")
(myvenv) PS C:\Users\Cristian\djangogirls> git add .
Your branch is up to date with 'origin/master'.

Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        new file:   blog/urls.py
        modified:   blog/views.py
        modified:   mysite/urls.py

(myvenv) PS C:\Users\Cristian\djangogirls> git commit -m "Chnaged the HTML for the site."
[master 63b1970] Chnaged the HTML for the site.
 4 files changed, 32 insertions(+)
 create mode 100644 blog/templates/blog/post_list.html
(myvenv) PS C:\Users\Cristian\djangogirls> git push
Enumerating objects: 15, done.
Counting objects: 100% (15/15), done.
Delta compression using up to 8 threads
Compressing objects: 100% (8/8), done.
Writing objects: 100% (10/10), 1.23 KiB | 157.00 KiB/s, done.
Total 10 (delta 4), reused 0 (delta 0), pack-reused 0 (from 0)
remote: Resolving deltas: 100% (4/4), completed with 4 local objects.
To https://github.com/CristianNumberThree/my-first-blog.git
   5667ba1..63b1970  master -> master
(myvenv) PS C:\Users\Cristian\djangogirls> python manage.py shell
Python 3.11.8 (tags/v3.11.8:db85d51, Feb  6 2024, 22:03:32) [MSC v.1937 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.        
(InteractiveConsole)
>>> Post.objects.all()
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 'Post' is not defined
>>> from blog.models import Post
>>> Post.objects.all()
<QuerySet [<Post: FirstBlogPost>, <Post: BlogPost2>, <Post: BlogPost3>, <Post: BlogPost4>, <Post: BlogPost5>, <Post: BlogPost6>, <Post: BlogPost7>]>        
>>> Post.objects.create(author=me, title='Sample title', text='Test') 
Traceback (most recent call last):
  File "<console>", line 1, in <module>
NameError: name 'me' is not defined
>>> from django.contrib.auth.models import User
>>> User.objects.all()
<QuerySet [<User: CristianNumber3>]>
>>> me = User.objects.get(username='ola')
Traceback (most recent call last):
  File "<console>", line 1, in <module>
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\db\models\manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\Cristian\djangogirls\myvenv\Lib\site-packages\django\db\models\query.py", line 435, in get
    raise self.model.DoesNotExist(
django.contrib.auth.models.User.DoesNotExist: User matching query does not exist.
>>> me = User.objects.get(username='CristianNumber3')
>>> Post.objects.create(author=me, title='Sample title', text='Test')
<Post: Sample title>
>>> Post.objects.all()
<QuerySet [<Post: FirstBlogPost>, <Post: BlogPost2>, <Post: BlogPost3>, <Post: BlogPost4>, <Post: BlogPost5>, <Post: BlogPost6>, <Post: BlogPost7>, <Post: Sample title>]>
>>> Post.objects.filter(author=me)
<QuerySet [<Post: FirstBlogPost>, <Post: BlogPost2>, <Post: BlogPost3>, <Post: BlogPost4>, <Post: BlogPost5>, <Post: BlogPost6>, <Post: BlogPost7>, <Post: Sample title>]>
>>> Post.objects.filter(title__contains='title')
<QuerySet [<Post: Sample title>]>
>>> from django.utils import timezone
>>> Post.objects.filter(published_date__lte=timezone.now())
<QuerySet [<Post: BlogPost6>, <Post: BlogPost7>]>
>>> post = Post.objects.get(title="Sample title")
>>> post.publish()
>>> Post.objects.filter(published_date__lte=timezone.now())
<QuerySet [<Post: BlogPost6>, <Post: BlogPost7>, <Post: Sample title>]>
>>> Post.objects.order_by('created_date')
<QuerySet [<Post: FirstBlogPost>, <Post: BlogPost2>, <Post: BlogPost3>, <Post: BlogPost4>, <Post: BlogPost5>, <Post: BlogPost6>, <Post: BlogPost7>, <Post: Sample title>]>
>>> Post.objects.order_by('-created_date')
<QuerySet [<Post: Sample title>, <Post: BlogPost7>, <Post: BlogPost6>, <Post: BlogPost5>, <Post: BlogPost4>, <Post: BlogPost3>, <Post: BlogPost2>, <Post: FirstBlogPost>]>
>>> Post.objects.filter(published_date__lte=timezone.now()).order_by('published_date')
<QuerySet [<Post: BlogPost6>, <Post: BlogPost7>, <Post: Sample title>]>
>>> exit()
(myvenv) PS C:\Users\Cristian\djangogirls> python manage.py runserver
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
February 25, 2024 - 22:37:42
Django version 3.2.24, using settings 'mysite.settings'
Starting development server at http://127.0.0.1:8000/  
Quit the server with CTRL-BREAK.
(myvenv) PS C:\Users\Cristian\djangogirls> python manage.py runserver
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
February 25, 2024 - 22:39:32
Django version 3.2.24, using settings 'mysite.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CTRL-BREAK.
[25/Feb/2024 22:39:36] "GET / HTTP/1.1" 200 821
(myvenv) PS C:\Users\Cristian\djangogirls> git status
On branch master
Your branch is up to date with 'origin/master'.

  (use "git restore <file>..." to discard changes in working directory)       
        modified:   blog/templates/blog/post_list.html
        modified:   blog/views.py

Untracked files:
  (use "git add <file>..." to include in what will be committed)
        Script module 5 vs code.txt

no changes added to commit (use "git add" and/or "git commit -a")
On branch master
Your branch is up to date with 'origin/master'.

  (use "git restore --staged <file>..." to unstage)
        new file:   Script module 5 vs code.txt
        modified:   blog/templates/blog/post_list.html
        modified:   blog/views.py

(myvenv) PS C:\Users\Cristian\djangogirls> git commit -m "Modified templates to display posts from database."
[master fa810b5] Modified templates to display posts from database.
 3 files changed, 271 insertions(+), 14 deletions(-)
(myvenv) PS C:\Users\Cristian\djangogirls> git push
Enumerating objects: 14, done.
Counting objects: 100% (14/14), done.
Delta compression using up to 8 threads
Compressing objects: 100% (6/6), done.
Writing objects: 100% (8/8), 3.76 KiB | 642.00 KiB/s, done.
Total 8 (delta 2), reused 0 (delta 0), pack-reused 0 (from 0)
remote: Resolving deltas: 100% (2/2), completed with 2 local objects.
To https://github.com/CristianNumberThree/my-first-blog.git
   63b1970..fa810b5  master -> master
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
February 25, 2024 - 22:44:18
Django version 3.2.24, using settings 'mysite.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CTRL-BREAK.
[25/Feb/2024 22:44:21] "GET / HTTP/1.1" 200 1031
(myvenv) PS C:\Users\Cristian\djangogirls> python manage.py runserver
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
February 25, 2024 - 22:45:09
Django version 3.2.24, using settings 'mysite.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CTRL-BREAK.
[25/Feb/2024 22:45:16] "GET / HTTP/1.1" 200 999
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
February 25, 2024 - 22:52:50
Django version 3.2.24, using settings 'mysite.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CTRL-BREAK.
[25/Feb/2024 22:52:53] "GET / HTTP/1.1" 200 1060
[25/Feb/2024 22:52:53] "GET /static/css/blog.css HTTP/1.1" 200 36
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
February 25, 2024 - 22:53:45
Django version 3.2.24, using settings 'mysite.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CTRL-BREAK.
[25/Feb/2024 22:53:48] "GET / HTTP/1.1" 200 1060
[25/Feb/2024 22:53:48] "GET /static/css/blog.css HTTP/1.1" 200 72
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
February 25, 2024 - 22:54:51
Django version 3.2.24, using settings 'mysite.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CTRL-BREAK.
[25/Feb/2024 22:54:54] "GET / HTTP/1.1" 200 1170
[25/Feb/2024 22:54:54] "GET /static/css/blog.css HTTP/1.1" 304 0
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
February 25, 2024 - 22:55:35
Django version 3.2.24, using settings 'mysite.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CTRL-BREAK.
[25/Feb/2024 22:55:38] "GET / HTTP/1.1" 200 1170
[25/Feb/2024 22:55:38] "GET /static/css/blog.css HTTP/1.1" 200 101
(myvenv) PS C:\Users\Cristian\djangogirls> python manage.py runserver
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
February 25, 2024 - 23:01:01
Django version 3.2.24, using settings 'mysite.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CTRL-BREAK.
[25/Feb/2024 23:01:04] "GET / HTTP/1.1" 200 1863
(myvenv) PS C:\Users\Cristian\djangogirls> python manage.py runserver
Watching for file changes with StatReloader
Performing system checks...

System check identified no issues (0 silenced).
February 25, 2024 - 23:09:07
Django version 3.2.24, using settings 'mysite.settings'
Starting development server at http://127.0.0.1:8000/
Quit the server with CTRL-BREAK.
[25/Feb/2024 23:09:20] "GET / HTTP/1.1" 200 2027
[25/Feb/2024 23:09:20] "GET /static/css/blog.css HTTP/1.1" 304 0
[25/Feb/2024 23:09:25] "GET / HTTP/1.1" 200 2027
(myvenv) PS C:\Users\Cristian\djangogirls> 